### PR TITLE
[MNOE-476] - Markup screens

### PR DESCRIPTION
**Pull request by fgourichon**
For review

### CODE REVIEW

File: api/app/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      if @product_markup.errors.empty?
+        render :show
+      else
+        render json: @product_markup.errors, status: :bad_request
+      end
+    end
+
+    # PATCH /mnoe/jpi/v1/admin/product_markups/:id
+    def update
+      # TODO: replace with authorize/ability
```
Comment (alachaum) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller.rb]: Any action required here?

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
+require 'rails_helper'
+
+module MnoEnterprise
+  describe Jpi::V1::Admin::ProductMarkupsController, type: :controller do
+    include MnoEnterprise::TestingSupport::SharedExamples::JpiV1Admin
+    # TODO: Fix Spec for Admin Controller
+    # before { skip }
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb]: Remove these two lines

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    # Stub user and user call
+    let(:user) { build(:user, admin_role: 'admin') }
+    let!(:current_user_stub) { stub_api_v2(:get, "/users/#{user.id}", user, %i(deletion_requests organizations orga_relations dashboards)) }
+    before do
+      sign_in user
+    end
+
+    describe 'GET #index' do
+
+
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb]: Remove extra blank lines

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      subject { get :index }
+
+      it_behaves_like 'a jpi v1 admin action'
+    end
+
+    describe 'GET #show' do
+
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb]: Remove extra blank line

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      subject { get :show, id: product_markup.id }
+
+      it_behaves_like 'a jpi v1 admin action'
+    end
+
+    describe 'POST #create' do
+      let(:params) { FactoryGirl.attributes_for(:product_markup) }
+      before { allow(MnoEnterprise::ProductMarkup).to receive(:create) { product_markup } }
+
+      subject { post :create, product_markup: params }
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller_spec.rb]: subject should be first

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/products_controller_spec.rb [hunk 6]
```diff
@@ -25,7 +25,7 @@ module MnoEnterprise
     describe 'GET #index' do
       let(:product) { build(:product) }
 
-      before { stub_api_v2(:get, '/products', [product], [:values, :assets, :categories, :product_pricings, :product_contracts], {}) }
+      before { stub_api_v2(:get, '/products', [product], [:'values.field', :assets, :categories, :product_pricings, :product_contracts], {}) }
 
       subject { get :index }
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/products_controller_spec.rb]: subject should be first

File: api/spec/controllers/mno_enterprise/jpi/v1/admin/products_controller_spec.rb [hunk 7]
```diff
@@ -35,7 +35,7 @@ module MnoEnterprise
     describe 'GET #show' do
       let(:product) { build(:product) }
 
-      before { stub_api_v2(:get, "/products/#{product.id}", product, [:values, :assets, :categories, :product_pricings, :product_contracts], {}) }
+      before { stub_api_v2(:get, "/products/#{product.id}", product, [:'values.field', :assets, :categories, :product_pricings, :product_contracts], {}) }
 
       subject { get :show, id: product.id }
```
Comment (alachaum) [inline api/spec/controllers/mno_enterprise/jpi/v1/admin/products_controller_spec.rb]: subject should be first

**Comment by fgourichon** [comment]
@alachaum, fixed

File: api/app/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller.rb [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    # GET /mnoe/jpi/v1/admin/product_markups/1
+    def show
+      @product_markup = MnoEnterprise::ProductMarkup.find_one(params[:id], :product, :organization)
+    end
+
+    # POST /mnoe/jpi/v1/admin/product_markups
+    def create
+      @product_markup = MnoEnterprise::ProductMarkup.new(product_markups_params)
+      @product_markup.relationships.product = MnoEnterprise::Product.new(id: product_markups_create_params[:product_id])
```
Comment (cesar-tonnoir) [inline api/app/controllers/mno_enterprise/jpi/v1/admin/product_markups_controller.rb]: Shouldn't that be in the initializer of your `MnoEnterprise::ProductMarkup` ?

File: api/config/routes.rb [hunk 9]
```diff
@@ -262,6 +262,7 @@
         end
 
         resources :app_metrics, only: [:index, :show]
+        resources :product_markups, only: [:index, :show, :destroy, :update, :create]
```
Comment (cesar-tonnoir) [inline api/config/routes.rb]: Routing spec missing

**Pull request merged**